### PR TITLE
Better Objective-C FFI and cleanup for OS X

### PR DESCRIPTION
**Pull request by xymus**
This PR opens the way for the iOS platform support.

Two commits may look very similar: "ObjC FFI: import public Objective-C headers for imported modules" and "nitc: fix duplication on imports of public foreign code". They both manage importing headers from other modules so public extern classes may work across module, but one is for Objective-C and the other for C. This service will have to be implemented for other languages too (mostly C++), I'll be looking for a higher level service at that time.

### CODE REVIEW

File: src/ffi/objc.nit [hunk 1]
```diff
@@ -32,6 +32,27 @@ end
...
+
+		# gather from importation
+		for m in in_importation.direct_greaters do
+			# does the super module has inherited dependencies?
+			var import_dep = m.objc_imported_headers
+			if not import_dep.is_empty then
+				dep.add_all import_dep
+			end
+
+			# does the super module itself has aa public header?
```
Comment (Morriar) [inline src/ffi/objc.nit]: a*

**Comment by Morriar** [comment]
+1

**Comment by xymus** [comment]
@MehdiAit and @Tagachi please take a look at this PR. Take a look at http://nitlanguage.org/internal/workflow/, especially the "Relecture de code" part, for our code review policy.

@BlackMinou please review too.

**Comment by R4PaSs** [comment]
+1

**Comment by xymus** [comment]
Rerolled for comments.

**Comment by xymus** [comment]
Rerolled to fix tests.

**Comment by Morriar** [comment]
+1

**Comment by privat** [comment]
+1

**Pull request merged**